data_and_models/pipelines/ner/clean.py

```diff
@@ -0,0 +1,121 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""
+This script cleans annotations exported with Prodigy as .jsonl files.
+
+Run 'analyze.py' before to identify relevant cleaning rules.
+
+Usage: clean.py [OPTIONS] INPUT_PATH KEEP_LABEL [RENAME_INTO]
+
+First, this script keeps only valid texts from the annotations (see next paragraph).
+Second, this script normalizes entity labels (use upper case for labels in lower case).
+Third, this script keeps only the given entity label and renames it if necessary.
+
+Valid texts are texts which have been accepted, have or not spans (i.e. entities), and
+have a unique 'input_hash'.
+
+The output is one file. The file is put in the same directory as the input file. It is
+named following the pattern 'annotations_<entity_label>.jsonl'.
+"""
+
+from collections import defaultdict
+from copy import deepcopy
```

Comment:
It appears that these two imports are unused.

Reply:
Done in 29795f22c5d2024f1d81ecc2fb312a7766a8dda6.

Resolving the discussion to make reading the reviews easier. Feel free to reopen if needed.